skipchain/struct.go

```diff
@@ -887,7 +887,11 @@ func (db *SkipBlockDB) getFromTx(tx *bolt.Tx, sbID SkipBlockID) (*SkipBlock, err
 		return nil, nil
 	}
 
-	_, sbMsg, err := network.Unmarshal(val, cothority.Suite)
+	// For some reason boltdb changes the val before Unmarshal finishes. When
```

Comment:
This is really important, and I just spent some time on it yesterday in OmniLedger as well. I wonder if we should put a wrapper on tx.Bucket.Get so that by default we always get a new copy. Gross, wasteful, but safe.

Reply:
A wrapper sounds good to me. It is a memory copy, and we do much worse (more time consuming) things with it after that, so I think copying everything from `Bucket.Get` should not slow us down a lot.

Reply:
Opened #1269